org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/proxy/ProxySettingsTest.groovy

```diff
@@ -0,0 +1,348 @@
+        firstJob.join()
+        setupTestProxyData(secondTempHost, 8080, userId, password)
+        secondJob.schedule()
+        secondJob.join()
+
+        then:
+        System.getProperty("http.proxyHost") == permHost
+        retrievedHost == secondTempHost
+    }
+
+    def "Proxies are accessed upon Gradle distribution download"() {
+        setup:
+        proxyServer.start()
+        proxyServer.requireAuthentication(userId, password)
+        setupTestProxyData("localhost", proxyServer.port, userId, password)
+        File rootProject = newProject(true, true)
+        server.expectGet("/gradlew/dist", new File(dependencyTempFolder.root, 'gradle-bin.zip'))
+        Job job = newProjectImportJob(rootProject)
+
+        when:
+        job.schedule()
+        job.join()
+
+        then:
+        proxyServer.requestCount == 1
+    }
+
+    def "Proxies are accessed from Gradle build VM"() {
+        setup:
+        proxyServer.start()
+        proxyServer.requireAuthentication(userId, password)
+        setupTestProxyData("localhost", proxyServer.port, userId, password)
+        File rootProject = newProject(false, true)
+        server.expectGet("/not-a-real-group/not-a-real-dependency/0.0/not-a-real-dependency-0.0.pom", new File(dependencyTempFolder.root, 'not-a-real-dependency-0.0.pom'))
+        def job = new RunGradleConfigurationDelegateJob(createLaunchMock(), createLaunchConfigurationMock(rootProject.absolutePath))
+
+        when:
+        job.schedule()
+        job.join()
+
+        then:
+        proxyServer.requestCount == 1
+    }
+
+    def "Proxies are not accessed from Gradle build VM with incorrect proxy settings"() {
+        setup:
+        proxyServer.start()
+        proxyServer.requireAuthentication(userId, password)
+        setupTestProxyData("localhost", proxyServer.port, userId, password + '-incorrect-password')
+        File rootProject = newProject(false, true)
+        def job = new RunGradleConfigurationDelegateJob(createLaunchMock(), createLaunchConfigurationMock(rootProject.absolutePath))
+
+        when:
+        job.schedule()
+        job.join()
+
+        then:
+        proxyServer.requestCount == 0
+    }
+
+    def setupTestProxyData(String host, int port, String userId, String password) {
+        String proxyHostMemory = System.getProperty("http.proxyHost")
+        String proxyPortMemory = System.getProperty("http.proxyPort")
+        String proxyUserMemory = System.getProperty("http.proxyUser")
+        String proxyPassMemory = System.getProperty("http.proxyPassword")
+
+
+        // Gets rid of The HTTP proxy is enabled in the preferences but disabled in the system settings
+        final IProxyData httpProxyData = proxyService.getProxyData(IProxyData.HTTP_PROXY_TYPE)
+
+        httpProxyData.setHost(host)
+        httpProxyData.setPort(port)
+        httpProxyData.setUserid(userId)
+        httpProxyData.setPassword(password)
+        proxyService.setProxyData((IProxyData[]) [
```

Comment:
It turns out that `proxyService.setProxyData` automatically sets the System settings given certain `IProxyService` configurations (under some configurations, it just wipes them out). Lines 245-248 and 262-273 are meant to combat this behaviour to ensure that _"Different proxy settings can be used by subsequent builds"_ succeeds.